### PR TITLE
Add references and nullability to the README

### PR DESCRIPTION
**Pull request by Totktonada**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -111,3 +111,91 @@ Quering schema field types (the order matches `get_names`):
...
+## References
+
+Named types are ones that have mandatory `name` field in its definition:
+record, fixed, enum.
+
+Named types can be referenced after the first definition (forward declarations
+are forbidden). Of course, primitive types can be referenced in any place of a
+schema.
+
+Note 1: A reference is a usage of a type (not a value), so the effect is like
```
Comment (Khatskevich) [inline README.md]: Not sure about that. I think that it is not true.

File: README.md [hunk 2]
```diff
@@ -111,3 +111,91 @@ Quering schema field types (the order matches `get_names`):
...
 avro_schema.get_types(schema)
 ```
+
+## References
+
+Named types are ones that have mandatory `name` field in its definition:
+record, fixed, enum.
+
+Named types can be referenced after the first definition (forward declarations
+are forbidden). Of course, primitive types can be referenced in any place of a
```
Comment (Khatskevich) [inline README.md]: Thay will be allowed soon.
Do you want to change the text afterward or make it an undocumented feature?

File: README.md (hunk 1 above)
Comment (Totktonada) [reply README.md]: Need to be clarified, sure.

File: README.md (hunk 2 above)
Comment (Totktonada) [reply README.md]: I think we need to describe the current state and change the description with new ehancements.

**Comment by Totktonada** [comment]
@Khatskevich Thanks for the review. All comments are fixed.

**Pull request merged**